benchmark/src/polyval.c

```
@@ -133,15 +121,7 @@ static void _polyval(const struct polyval_key *key, const void *src,
 {
 	struct polyval_state polystate;
 	polyval_init(&polystate);
-
-	// Pad partial blocks since polyval can only handle 16-byte multiples.
-	u8 padded_final[POLYVAL_BLOCK_SIZE];
-	size_t remainder = srclen % POLYVAL_BLOCK_SIZE;
-	if (remainder) {
-		memset(padded_final, 0, POLYVAL_BLOCK_SIZE);
-		memcpy(&padded_final, src + srclen - remainder, remainder);
-	}
-	polyval_update(&polystate, key, src, srclen, padded_final, simd);
+	polyval_update(&polystate, key, src, srclen, simd);
```

There probably should be an assertion in this function: `ASSERT(srclen % POLYVAL_BLOCK_SIZE == 0);`.  We wouldn't want someone to run e.g. `cipherbench --bufsize=63` and have the inputs silently truncated.